api/pkg/shared/resource/resource_test.go

```diff
@@ -142,6 +142,26 @@ func TestQuery_ByTags(t *testing.T) {
 	assert.Equal(t, 1, len(res))
 }
 
+func TestQuery_ByCatalogs(t *testing.T) {
+	tc := testutils.Setup(t)
+	testutils.LoadFixtures(t, tc.FixturePath())
+
+	req := Request{
+		Db:    tc.DB(),
+		Log:   tc.Logger("resource"),
+		Name:  "",
+		Kinds: []string{},
+		Catalogs:  []string{"catalog-community"},
+		Limit: 100,
+	}
+
+	res, err := req.Query()
+	assert.NoError(t, err)
+
+	assert.Equal(t, 1, len(res))
+	assert.Equal(t,"catalog-community",res[0].Catalog.Name)
+}
+
```

Comment:
I think we should also add a negative test for this where we don't find any resource